server/app.js

```diff
@@ -11,7 +11,9 @@ app.use(express.urlencoded({extended: true}));
 
 app.get('/users/:userId', async (req, res) => {
   try {
-    const user = await getUserById(req.params.userId);
+    const userInfo = await getUserById(req.params.userId)
+    const languageInfo = await getUserLanguagesByUserId(req.params.userId);
+    let user = formatData(userInfo, languageInfo);
```

Comment:
instead of performing the lookup of `userId` in each function invocation, you could assign it to a variable, check if it exists first, and then pass it into each of your funcs here

Reply:
getUserById gets all the information except languages from the Users table. The getUserLanguagesByUserId does a query on the join tables and get the languages spoken by the user. Then formatData basically merges the 2 and sends an object to the client.